<launch>
    <arg name="input" default="/spencer/perception_internal/detected_persons/rgbd_front_top/upper_body"/>
    <arg name="output" default="/spencer/perception_internal/detected_persons_syncronized/rgbd_front_top/upper_body"/>

    <arg name="dlib_model_path" default="$(find openface_classifier)/config/models/dlib/shape_predictor_68_face_landmarks.dat" />
    <arg name="openface_model_path" default="$(find openface_classifier)/config/models/openface/nn4.small2.v1.t7" />

    <arg name="classifier_path" default="$(find openface_classifier)/config/models/classifier.pkl" />
    <arg name="feature_path" default="$(find openface_classifier)/config/models/features.npy" />
    <arg name="labels_path" default="$(find openface_classifier)/config/models/labels.npy" />

    <!-- Re-publish detections as syncronized track, for processing by the classifier-->
    <group ns="/spencer/perception_internal/detected_person_association">
        <node name="track_synchronizer" type="track_synchronizer.py" pkg="spencer_detected_person_association">
            <remap from="input" to="$(arg input)"/>
            <remap from="output" to="$(arg output)"/>
        </node>
    </group>

    <!-- <arg name="image_topic" default="/head_camera/rgb/image_rect_color"/> -->
    <arg name="image_topic" default="/hsrb/head_rgbd_sensor/rgb/image_rect_color"/>
    <arg name="annotations_topic" default="/spencer/perception_internal/people_detection/rgbd_front_top/upper_body_detector/annotations"/>
    <arg name="tracks_topic" default="/spencer/perception_internal/detected_persons_syncronized/rgbd_front_top/upper_body"/>

        <!-- Re-publish detections as syncronized track, for processing by the classifier-->
    <!-- <group ns="/spencer/classification_internal/classified_tracks">
        <node name="annotation_classifier" type="annotation_classifier_node.py" pkg="openface_classifier" output="screen">
            <remap from="image" to="$(arg image_topic)"/>
            <remap from="annotations" to="$(arg annotations_topics)"/>
            <remap from="tracks" to="$(arg tracks_topics)"/>
        </node>
    </group> -->

    <!-- Re-publish detections as syncronized track, for processing by the classifier-->
    <group ns="/spencer/classification_internal/classified_tracks">
        <node name="face_embedder" type="face_embedder_node.py" pkg="openface_classifier">
            <param name="dlib_model_path" value="$(arg dlib_model_path)" type="string"/>
            <param name="openface_model_path" value="$(arg openface_model_path)" type="string"/>

            <param name="image_topic" value="$(arg image_topic)" type="string"/>
            <param name="annotations_topic" value="$(arg annotations_topic)" type="string"/>
            <param name="tracks_topic" value="$(arg tracks_topic)" type="string"/>
            <param name="embeddings_topic" value="embeddings" type="string"/>
        </node>


        <node name="classifier" type="classifier_node.py" pkg="openface_classifier" output="screen">
            <param name="classifier_path" value="$(arg classifier_path)" type="string"/>
            <param name="feature_path" value="$(arg feature_path)" type="string"/>

            <param name="embeddings_topic" value="embeddings" type="string"/>
            <param name="labeled_tracks_topic" value="labeled_tracks" type="string"/>
        </node>

        <node name="trainer" type="trainer_node.py" pkg="openface_classifier" output="screen">
            <param name="classifier_path" value="$(arg classifier_path)" type="string"/>
            <param name="feature_path" value="$(arg feature_path)" type="string"/>
            <param name="labels_path" value="$(arg labels_path)" type="string"/>
        </node>

        <node name="data_associator" type="data_associator_node.py" pkg="openface_classifier" output="screen">
            <param name="embeddings_topic" value="embeddings" type="string"/>
        </node>

        <node name="call_person" type="call_person_by_name.py" pkg="openface_classifier" output="screen">
            <param name="labeled_tracks_topic" value="labeled_tracks" type="string"/>
            <param name="call_person_topic" value="call_person" type="string"/>
        </node>
    </group>

</launch>
